x-pack/plugins/observability_solution/slo/server/services/transform_generators/transform_generator.ts

```diff
@@ -76,23 +51,16 @@ export abstract class TransformGenerator {
 
     const groupings =
       !groups.includes(ALL_VALUE) && groups.length
-        ? groups.reduce(
-            (acc, field) => {
-              return {
-                ...acc,
-                [`slo.groupings.${field}`]: {
-                  terms: {
-                    field,
-                  },
+        ? groups.reduce((acc, field) => {
+            return {
+              ...acc,
+              [`slo.groupings.${field}`]: {
+                terms: {
+                  field,
                 },
-              };
-            },
-            {
-              'slo.instanceId': {
-                terms: { field: 'slo.instanceId' },
               },
-            }
-          )
+            };
+          }, {})
         : { 'slo.instanceId': { terms: { field: 'slo.instanceId' } } };
```

Comment:
```suggestion
        : { };
```

Reply:
This will fix the issue with the SLOs without the group by. Since we are no longer setting the `slo.instanceId` runtime field, there is nothing for the transform to pivot on so it's not generating any documents.